src/compiler/app-core/app-data.ts

```diff
@@ -24,23 +24,13 @@ export const getBuildFeatures = (cmps: ComponentCompilerMeta[]): BuildFeatures =
   const slotRelocation = cmps.some((c) => c.encapsulation !== 'shadow' && c.htmlTagNames.includes('slot'));
   const f: BuildFeatures = {
     allRenderFn: cmps.every((c) => c.hasRenderFn),
-    cmpDidLoad: cmps.some((c) => c.hasComponentDidLoadFn),
-    cmpShouldUpdate: cmps.some((c) => c.hasComponentShouldUpdateFn),
     cmpDidUnload: cmps.some((c) => c.hasComponentDidUnloadFn),
```

Comment:
Why do we keep this one?

Reply:
it's legacy and has been deprecated (but not removed) since v2.. I didn't know whether to nuke it / start pulling it out as part of this effort or just do it all in one swoop. Happy to just take it out of this bit

Reply:
Let's remove it, I don't see it anywhere mentioned in the docs or somewhere else.

Same for `[hasLifecycle](https://github.com/ionic-team/stencil/pull/6167/commits/916bc33cdc1fcd22eb76ed2f13080eb3fcfd3e94)`?

Reply:
component `hasLifecycle` flags are used to to turn on the build flag `lifecycle` which in turn is used to turn on the flag `member` which is used quite a lot for runtime stuff